File: internal/model/picture.go
package model

import (
	"time"

	"github.com/google/uuid"
)

type Picture struct {
	ID           uuid.UUID
	Name         string
	Size         PictureSize
	ScrapeInfo   ScrapeInfo
	Tags         []string
	GenerateInfo GenerateInfo
}

type PictureSize struct {
	Width  int
	Height int
}

type ScrapeInfo struct {
	Time     time.Time
	Path     string
	FileHash string
}

type GenerateInfo struct {
	Prompt         string
	NegativePrompt string
	Steps          int
	Size           string
	Seed           string
	Sampler        string
	CfgScale       float64
	Model          GenerateModel
	Additional     map[string]string
}

type GenerateModel struct {
	Name string
	Hash string
}

File: frontend/src/Features/PictureView/Models/Picture.ts
export type PictureSize = {
    width: number,
    height: number
};

export type PictureGenerateInfo = {
    prompt: string,
    negativePrompt: string,
    steps: number,
    size: string,
    seed: string,
    sampler: string,
    cfgScale: number,
    modelHash: string
};

export type Picture = {
    id: string,
    size: PictureSize,
    tags: string[],
    info: PictureGenerateInfo
};